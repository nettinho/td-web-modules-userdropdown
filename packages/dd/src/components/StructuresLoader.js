import React from "react";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Loading } from "@truedat/core/components";
import { clearStructures, fetchStructures } from "../routines";

export class StructuresLoader extends React.Component {
  static propTypes = {
    fetchStructures: PropTypes.func,
    clearStructures: PropTypes.func,
    structuresLoading: PropTypes.bool
  };

  componentDidMount() {
    const { fetchStructures } = this.props;
    fetchStructures({});
  }

  componentWillUnmount() {
    const { clearStructures } = this.props;
    clearStructures();
  }

  render() {
    const { structuresLoading } = this.props;

    if (structuresLoading) {
      return <Loading />;
    } else {
      return null;
    }
  }
}

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ clearStructures, fetchStructures }, dispatch)
});

const mapStateToProps = ({ structuresLoading }) => ({ structuresLoading });

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(StructuresLoader);
