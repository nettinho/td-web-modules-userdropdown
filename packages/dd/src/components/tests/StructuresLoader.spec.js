import React from "react";
import { shallow } from "enzyme";
import { StructuresLoader } from "../StructuresLoader";

describe("<StructuresLoader />", () => {
  it("matches the latest snapshot", () => {
    const fetchStructures = jest.fn();
    const structuresLoading = true;
    const props = { fetchStructures, structuresLoading };
    const wrapper = shallow(<StructuresLoader {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a loader if structuresLoading is true", () => {
    const fetchStructures = jest.fn();
    const structuresLoading = true;
    const props = { fetchStructures, structuresLoading };
    const wrapper = shallow(<StructuresLoader {...props} />);
    expect(wrapper.find("Loading").length).toBe(1);
  });

  it("renders null if structuresLoading is false", () => {
    const fetchStructures = jest.fn();
    const structuresLoading = false;
    const props = { fetchStructures, structuresLoading };
    const wrapper = shallow(<StructuresLoader {...props} />);
    expect(wrapper.getElement()).toBeNull();
  });

  it("calls fetchStructures when component mounts, clearStructures when component unmounts", () => {
    const fetchStructures = jest.fn();
    const clearStructures = jest.fn();
    const structuresLoading = false;
    const props = { fetchStructures, structuresLoading, clearStructures };
    jest.spyOn(StructuresLoader.prototype, "componentDidMount");
    const wrapper = shallow(<StructuresLoader {...props} />);
    expect(StructuresLoader.prototype.componentDidMount.mock.calls.length).toBe(
      1
    );
    expect(clearStructures.mock.calls.length).toBe(0);
    expect(fetchStructures.mock.calls.length).toBe(1);
    wrapper.unmount();
    expect(clearStructures.mock.calls.length).toBe(1);
    expect(fetchStructures.mock.calls.length).toBe(1);
  });
});
