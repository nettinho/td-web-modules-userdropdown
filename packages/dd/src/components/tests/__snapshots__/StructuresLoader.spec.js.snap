// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StructuresLoader /> matches the latest snapshot 1`] = `<Loading />`;
